### PR TITLE
[SYCLomatic] Support 33 cublas API migration

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei zhiwei.jiang@intel.com

### CODE REVIEW

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/runtime/dpct-rt/include/dpct/blas_utils.hpp [hunk 1]
```diff
@@ -538,9 +543,80 @@ inline void rot_impl(sycl::queue &q, int n, void *x, int incx, void *y,
...
+template <class T, bool is_max>
+inline void iamaxmin_impl(sycl::queue &q, std::int64_t n, const void *x,
+                          std::int64_t incx, std::int64_t *res) {
+#ifndef __INTEL_MKL__
+  throw std::runtime_error("The oneAPI Math Kernel Library (oneMKL) Interfaces "
+                           "Project does not support this API.");
+#else
+  auto data_x = get_memory<T>(x);
+  auto data_res = get_memory<std::int64_t>(res);
+  if constexpr (is_max)
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/blas_utils.hpp]: use 2 functions

File: clang/runtime/dpct-rt/include/dpct/blas_utils.hpp [hunk 2]
```diff
@@ -538,9 +543,80 @@ inline void rot_impl(sycl::queue &q, int n, void *x, int incx, void *y,
...
+                      std::int64_t incx, void *res) {
+#ifndef __INTEL_MKL__
+  throw std::runtime_error("The oneAPI Math Kernel Library (oneMKL) Interfaces "
+                           "Project does not support this API.");
+#else
+  auto data_x = get_memory<Tx>(x);
+  auto data_res = get_memory<Tres>(res);
+  oneapi::mkl::blas::column_major::asum(q, n, data_x, incx, data_res);
+#endif
+}
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/blas_utils.hpp]: Can we add a macro to unify?

File: clang/runtime/dpct-rt/include/dpct/blas_utils.hpp [hunk 3]
```diff
@@ -2283,30 +2211,511 @@ inline void trmm(descriptor_ptr desc_ptr, oneapi::mkl::side left_right,
...
+/// vector.
+/// \tparam is_max True means finding the the largest absolute value index.
+/// \param [in] desc_ptr Descriptor.
+/// \param [in] n Number of elements in vector x.
+/// \param [in] x Input vector x.
+/// \param [in] x_type Data type of the vector x.
+/// \param [in] incx Stride of vector x.
+/// \param [out] result The index of the maximal/minimum element.
+template <bool is_max>
+inline void iamaxmin(descriptor_ptr desc_ptr, std::int64_t n, const void *x,
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/blas_utils.hpp]: use 2 functions

File: clang/runtime/dpct-rt/include/dpct/blas_utils.hpp [hunk 4]
```diff
@@ -2283,30 +2211,511 @@ inline void trmm(descriptor_ptr desc_ptr, oneapi::mkl::side left_right,
...
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/blas_utils.hpp]: use 2 functions

**Review by ziranzha — APPROVED** [review]

File: clang/runtime/dpct-rt/include/dpct/blas_utils.hpp (hunk 2 above)
Comment (zhiweij1) [reply clang/runtime/dpct-rt/include/dpct/blas_utils.hpp]: no changes for now.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**